### PR TITLE
refactor(enhancers): cleanup enhancers js logic, resolve css bug

### LINKED ISSUE #5308: enhancers z index issue
image

### PR DESCRIPTION
**Pull request by josephmcg**
### What this PR does 📖
- title

### Which issue(s) this PR fixes 🔨
- Resolve #5308 

### Special notes for reviewers 🗒️
- enhancers still needs a rework, some of this is kinda weird

### Additional comments 🎤
-

### CODE REVIEW

**Comment by molimauro** [comment]
Yes also the enhancers on the edit chatbar are not working.
It's weird to have the <Enhancers> in the Chatbar.vue and trigger it with vuex also from outside (like from the Editable in the Edit.html). Furthermore, by using `setChatbarContent` we broke the emoji insertion in the middle of a string, we always insert the new emojy at the end even if the cursor is not at the end. We should use `handleChatTextFromOutside` instead

**Comment by josephmcg** [comment]
Agreed, i can take a look at it tomorrow

**Comment by josephmcg** [comment]
got sidetracked working on other bugs. maybe merging this first would be good so the PR isnt too large in scale

File: components/interactables/ButtonGroup/ButtonGroup.vue [hunk 1]
```diff
@@ -1,83 +1,9 @@
...
-          if (this.action != null) {
-            this.action(text)
-          }
-        })
-      })
-    },
-  },
-})
-</script>
+<script setup lang="ts"></script>
```
Comment (molimauro) [inline components/interactables/ButtonGroup/ButtonGroup.vue:7]: Why is this empty?

Comment (josephmcg) [reply components/interactables/ButtonGroup/ButtonGroup.vue:7]: i assumed it was needed, but I'm not sure. can try removing

Comment (josephmcg) [reply components/interactables/ButtonGroup/ButtonGroup.vue:7]: they were very heavy-handed. I opted to placed click listeners on the buttons to achieve the same functionality

Comment (molimauro) [reply components/interactables/ButtonGroup/ButtonGroup.vue:7]: Ahaha yeah i noticed. So we're good. Can we remove the props from the component in `components/views/chat/enhancers/Enhancers.html` then right?

Comment (josephmcg) [reply components/interactables/ButtonGroup/ButtonGroup.vue:7]: ah yes, thanks

**Review by molimauro — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**